packages/playwright-core/src/grid/githubGridFactory.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GridAgentLaunchOptions, GridFactory } from './gridServer';
+import https from 'https';
+import debug from 'debug';
+
+const repoName = process.env.GITHUB_AGENT_REPO;
+if (!repoName)
+  throw new Error('GITHUB_AGENT_REPO is not specified.');
+
+const repoAccessToken = process.env.GITHUB_AGENT_REPO_ACCESS_TOKEN;
+if (!repoAccessToken)
+  throw new Error('GITHUB_AGENT_REPO_ACCESS_TOKEN is not specified.');
+
+const log = debug(`pw:grid:server`);
+
+const githubFactory: GridFactory = {
+  name: 'Agents hosted on Github',
+  capacity: 10,
+  launchTimeout: 10000,
```

Comment:
These two timeouts seem too short :)

Reply:
Increased, but launch timeout in this case is just time to trigger workflow (without waiting for it to finish etc).